### PR TITLE
Update pbs-gasnetrun_ibv launcher

### PR DESCRIPTION
**Pull request by jhh67**
Modify to work with "place/select" qsub syntax

Resolves https://github.com/Cray/chapel-private/issues/1800 and fixes https://github.com/chapel-lang/chapel/issues/20685.

Signed-off-by: John H. Hartman <jhh67@users.noreply.github.com>

### CODE REVIEW

File: runtime/src/launch/pbs-gasnetrun_ibv/launch-pbs-gasnetrun_ibv.c [hunk 1]
```diff
@@ -69,7 +70,7 @@ static qsubVersion determineQsubVersion(void) {
 
   if (strstr(version, "NCCS")) {
     return nccs;
-  } else if (strstr(version, "PBSPro")) {
+  } else if (strstr(version, "pbs_version") || strstr(version, "PBSPro")) {
```
Comment (ronawho) [inline runtime/src/launch/pbs-gasnetrun_ibv/launch-pbs-gasnetrun_ibv.c:73]: I'm not sure if it's worthwhile, but if we want to maintain compatibly with the old mpp style syntax, I'd detect `pbs_version` as `pbspro_select` or the older one as `pbspro_mpp` and specialize for them

**Comment by ronawho** [comment]
This will resolve https://github.com/Cray/chapel-private/issues/1800, and I've verified it'll fix https://github.com/chapel-lang/chapel/issues/20685. I'd expect it to resolve https://github.com/chapel-lang/chapel/issues/18567 as well

File: runtime/src/launch/pbs-gasnetrun_ibv/launch-pbs-gasnetrun_ibv.c [hunk 2]
```diff
@@ -231,6 +236,15 @@ int chpl_launch(int argc, char* argv[], int32_t numLocales) {
...
 int chpl_launch_handle_arg(int argc, char* argv[], int argNum,
                            int32_t lineno, int32_t filename) {
+
+  if (!strcmp(argv[argNum], CHPL_WALLTIME_FLAG)) {
+    walltime = argv[argNum+1];
+    return 2;
+  } else if (!strncmp(argv[argNum], CHPL_WALLTIME_FLAG"=", strlen(CHPL_WALLTIME_FLAG))) {
+    walltime = &(argv[argNum][strlen(CHPL_WALLTIME_FLAG)+1]);
+    return 1;
+  }
```
[the remark this answers is not in the document]
Comment (jhh67) [reply runtime/src/launch/pbs-gasnetrun_ibv/launch-pbs-gasnetrun_ibv.c:247]: Done.

File: runtime/src/launch/pbs-gasnetrun_ibv/launch-pbs-gasnetrun_ibv.c (hunk 1 above)
Comment (jhh67) [reply runtime/src/launch/pbs-gasnetrun_ibv/launch-pbs-gasnetrun_ibv.c]: Fixed.

**Review by ronawho — APPROVED** [review]
FYI normal multi-locale paratest was clean for me with https://github.com/chapel-lang/chapel/pull/23186:

```sh
$CHPL_HOME/util/test/paratest.server -nodepara 6 -dirs release/examples -dirs runtime/configMatters
...
[Summary: #Successes = 266 | #Failures = 0 | #Futures = 0]
```

**Pull request merged**